core/src/main/java/hudson/Functions.java

```diff
@@ -956,11 +956,13 @@ public static String getFooterURL() {
             if (d.getGlobalConfigPage()==null)  continue;
 
             if (d instanceof GlobalConfiguration) {
-                if (predicate.apply(((GlobalConfiguration)d).getCategory()))
+                if (predicate.apply((d.getCategory()))) {
                     r.add(new Tag(c.ordinal(), d));
+                }
             } else {
-                if (predicate.apply(GlobalConfigurationCategory.get(Unclassified.class)))
+                if (predicate.apply(d.getCategory())) {
                     r.add(new Tag(0, d));
```

Comment:
Why `0`? And if `getCategory` has been pushed up into `Descriptor`, why are we still checking `instanceof GlobalConfiguration`? The special-case call check of `Unclassified` was only necessary because that was the implied category of non-`GlobalConfigurationCategory` `Descriptor`s, but now these can actually declare their category. I would think the whole logic could be simplified to

```java
if (predicate.apply(d.getCategory())) {
    r.add(new Tag(c.ordinal(), d));
}
```